### PR TITLE
Add update_user_password method to UserManagement module

### PR DESCRIPTION
**Pull request by adamzinder**
## Description

* Adds update_user_password to method to module and returns a user object back up the updated user.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: lib/workos/user_management.rb [hunk 1]
```diff
@@ -275,6 +275,33 @@ def verify_email(magic_auth_challenge_id:, code:)
 
         WorkOS::User.new(response.body)
       end
+
+      # Updates user user password.
+      #
+      # @param [String] id The unique ID of the User.
+      # @param [String] new_password The new password to set for the user.
```
Comment (willmanduffy) [inline lib/workos/user_management.rb:282]: Looks like this should be `password`

```suggestion
      # @param [String] password The new password to set for the user.
```

File: lib/workos/user_management.rb [hunk 2]
```diff
@@ -275,6 +275,33 @@ def verify_email(magic_auth_challenge_id:, code:)
...
+      sig do
+        params(
+          id: String,
+          password: String,
+        ).returns(WorkOS::User)
+      end
+      def update_user_password(id:, password:)
+        response = execute_request(
+          request: put_request(
+            path: "/users/#{id}",
```
Comment (willmanduffy) [inline lib/workos/user_management.rb:295]: In the Node SDK the URL is "/users/#{id}/password"

🤔

Comment (adamzinder) [reply lib/workos/user_management.rb]: fixed 👍

File: spec/support/fixtures/vcr_cassettes/user_management/update_user_password/valid.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+      - __cf_bm=J3vNmHK1YPJVev.rkMAZ9FTLeJzcHKsk3guGQi775Ho-1693238179-0-AUdn0HNzvr6HUmQ4AXUyJZ61yHDc4Xs+XSI4hvhcGZ9jxsCJD4JhJmknzcIY4wELXZDlnn2QD2+EOsLZVoGsVqI=;
+        path=/; expires=Mon, 28-Aug-23 16:26:19 GMT; domain=.workos.com; HttpOnly;
+        Secure; SameSite=None
+      - __cfruid=43d140b605df7e6da0318c6a7d599e333d850c2f-1693238179; path=/; domain=.workos.com;
+        HttpOnly; Secure; SameSite=None
+      Server:
+      - cloudflare
+    body:
+      encoding: ASCII-8BIT
+      string: '{"object":"user","id":"user_01H7TVSKS45SDHN5V9XPSM6H44","email":"willman@blips.app","email_verified":true,"first_name":null,"last_name":null,"created_at":"2023-08-14T20:28:58.929Z","updated_at":"2023-08-28T15:56:19.798Z","user_type":"unmanaged","email_verified_at":"2023-08-22T11:18:01.850Z","google_oauth_profile_id":null,"microsoft_oauth_profile_id":null}'
```
Comment (willmanduffy) [inline spec/support/fixtures/vcr_cassettes/user_management/update_user_password/valid.yml:79]: Can we edit my email out of the fixture? 😓

Comment (adamzinder) [reply spec/support/fixtures/vcr_cassettes/user_management/update_user_password/valid.yml]: done 👍

**Review by willmanduffy — APPROVED** [review]

**Pull request merged**